fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/PermissiveAvailabilitySpec.groovy

```diff
@@ -0,0 +1,145 @@
+        given:
+        interval1 = new Interval('2010-01-01/2020-12-31')
+        interval2 = new Interval('2017-01-01/2020-12-31')
+
+        permissiveAvailability = new PermissiveAvailability(
+                tableName,
+                [column1, column2] as Set,
+                new TestDataSourceMetadataService([
+                        (column1): [interval1] as Set,
+                        (column2): [interval2] as Set
+                ])
+        )
+
+        expect:
+        permissiveAvailability.getAllAvailableIntervals() == [
+                (column1): [interval1] as Set,
+                (column2): [interval2] as Set
+        ] as LinkedHashMap
+    }
+
+    @Unroll
+    def "getAvailableIntervals returns the union of requested columns with available intervals - #firstInterval & #secondInterval"() {
+        given:
+        interval1 = new Interval(firstInterval)
+        interval2 = new Interval(secondInterval)
+
+        permissiveAvailability = new PermissiveAvailability(
+                tableName,
+                [column1, column2] as Set,
+                new TestDataSourceMetadataService([
+                        (column1): [interval1] as Set,
+                        (column2): [interval2] as Set
+                ])
+        )
+
+        DataSourceConstraint dataSourceConstraint = Mock(DataSourceConstraint)
+        dataSourceConstraint.getAllColumnNames() >> ['column1', 'column2']
+
+        expect:
+        permissiveAvailability.getAvailableIntervals(dataSourceConstraint) == new SimplifiedIntervalList(
+                expected.stream()
+                        .map{it -> new Interval(it)}
+                        .collect(Collectors.toSet())
+        )
+
+        where:
+        firstInterval           | secondInterval          | expected
+        '2017-01-01/2017-02-01' | '2017-01-01/2017-02-01' | ['2017-01-01/2017-02-01']                          // full overlap (start/end, start/end)
+        '2017-01-01/2017-02-01' | '2018-01-01/2018-02-01' | ['2017-01-01/2017-02-01', '2018-01-01/2018-02-01'] // 0 overlap (-10/-1, 0/10)
+        '2017-01-01/2017-02-01' | '2017-02-01/2017-03-01' | ['2017-01-01/2017-03-01']                          // 0 overlap abutting (-10/0, 0/10)
+        '2017-01-01/2017-02-01' | '2017-01-15/2017-03-01' | ['2017-01-01/2017-03-01']                          // partial front overlap (0/10, 5/15)
+        '2017-01-01/2017-02-01' | '2016-10-01/2017-01-15' | ['2016-10-01/2017-02-01']                          // partial back overlap (0/10, -5/5)
+        '2017-01-01/2017-02-01' | '2017-01-15/2017-02-01' | ['2017-01-01/2017-02-01']                          // full front overlap (0/10, 5/10)
+        '2017-01-01/2017-02-01' | '2017-01-01/2017-01-15' | ['2017-01-01/2017-02-01']                          // full back overlap (0/10, 0/5)
+        '2017-01-01/2017-02-01' | '2017-01-15/2017-01-25' | ['2017-01-01/2017-02-01']                          // fully contain (0/10, 3/9)
```

Comment:
Some of the other stuff that's in that example shows off how Groovy will let you use strings as method names :-)